planner/core/planbuilder.go

```diff
@@ -4238,6 +4248,12 @@ func (b *PlanBuilder) buildLoadData(ctx context.Context, ld *ast.LoadDataStmt) (
 	if err != nil {
 		return nil, err
 	}
+
+	if detached {
```

Comment:
can we move this into `buildLoadData`, so no need to init `detached

Reply:
This is already buildLoadData. You mean buildLoadData (executor)?

executor builder can not change the schema of plan, it's one way

Reply:
I moved it into buildLoadData (executor) and the query schema is not changed. Maybe other components have already read the old value and decided the query output.